### ISSUE TITLE
does harbor support audit logs

### ISSUE THREAD

**Issue by shahidv3** [goharbor/harbor#15134]
hi All,

does harbor support audit logs. I tried searching in /var/log/harbor.
I want to track who has updated/added the configuration settings in project library.

Please let me know if there is an option for this

thanks

**Comment by shahidv3** [comment]
hi @reasonerjt  @bitsf 

can you please here

thanks

**Comment by bitsf** [comment]
Which audit logs do you suggest to support ?

**Comment by shahidv3** [comment]
Hi @bitsf 

My use case is if admin performs any change in project configuration settings like enable or disabling of content trust or preventing vulnerabilities images to run or update CVE list . Will it be captured in any logs .

 If not it would be great if this feature gets added / captured in proxy.logs or core.logs

please let me if these example actions are audited in logs or events now?


Thanks, sahid

**Comment by zyyw** [comment]
we need to re-triage it.

**Comment by qnetter** [comment]
Has recent work for internal user advanced this issue?

**Comment by ccunning** [comment]
I agree that Harbor needs more robust audit logging for security and compliance reasons. When making changes to the administration or project configuration, that should trigger an audit log.

I think it makes sense to have audit events for (most) POST, PUT, PATCH, and DELETE API requests, as that would be any changes to configuration. At the least, I would think:
- Create / Modify / Delete users
- Create / Modify / Delete groups
- Create / Modify / Delete robot accounts
- Modify system configurations
- Modify project membership
- Create / Modify / Delete project policies (retention and immutability)
- Modify auditlog configuration (scheduled cleanup, etc)

We are running into issues with compliance (especially around robot accounts) because, while some of these events fire an event, they don't get stored in a central log for auditing later.

**Comment by zyyw** [comment]
One more requirement for audit log from the community:
* https://github.com/goharbor/harbor/issues/19160

**Comment by slushysnowman** [comment]
This really needs to be added.

We just went through an exercise of trying to track down a chain of events for an incident, and in the end we had to look at ingress logs, and restore DB snapshots to be able to determine the chain of events.

and even then we couldn't determine WHO made the changes, the information just isn't present in our logging.

Ideally it'd be great if all POST/PUT/PATCH/DELETE actions were logged in an audit log, along with the identity of the caller.

Seriously, for enterprise use-cases this is **really really** important.

**Comment by OrlinVasilev** [comment]
@msha01 for visibility and to be added for 2.9/2.10/2.X :)
@yanji09 @wy65701436 any idea it was moved to 2.10 ?

**Comment by OrlinVasilev** [comment]
@msha01 for visibility and to be added for 2.9/2.10/2.X :)
@yanji09 @wy65701436 any idea it was moved to 2.10 ? 

anyone ?

**Comment by yanji09** [comment]
It is on Harbor's radar. But it was not in 2.10. Any contribution to this from community are welcome.
